contributing/guide-adding-external-secret-stores.md

```diff
@@ -0,0 +1,132 @@
+# Adding Secret Store Support
```

Comment:
Are we planning to drop these from the docs site? I definitely don’t think we want to maintain them in two separate places, but if I was developing a provider I’d likely think to look at the website rather the the Crossplane repo.

Reply:
I can't find the reference but @negz and I discussed where "contributing docs" for crossplane/crossplane live, and the answer was this repo. This content will be removed from the docs with v1.11 and later.

Reply:
Thanks for the context! @negz since the revert does result in the files being marked as moved do you think it still makes sense to bring the full directory back then delete it again?

Reply:
It seems like the approach I took does maintain the file history, despite it all showing green here, so yes.

Reply:
> If I was developing a provider I’d likely think to look at the website rather the the Crossplane repo.

It occurred to me after we made this decision that one (and a half) of these documents are mostly aligned with provider development. I'd still prefer to keep everything here for the time being - we can change later if we find there's a discoverability issue.

In particular I do think we want a `CONTRIBUTING.md` file in the repo where the code is, and that makes me want to keep everything "how to contribute code to the project" related in one place.

Reply:
@plumbis perhaps the docs site could have a very simple page for Contributing that then simply points at c/c/contributing?

Reply:
> It seems like the approach I took does maintain the file history, despite it all showing green here, so yes.

Hmm yeah looks like it updated since last time I was looking 🤔 

> In particular I do think we want a CONTRIBUTING.md file in the repo where the code is, and that makes me want to keep everything "how to contribute code to the project" related in one place.

I agree on `CONTRIBUTING.md` as much of what it is describing is how to contribute to _this_ repository. The others feel less appropriate, but it sounds like y'all have already had discussion around this so I am happy to defer 👍🏻

Reply:
Already ahead of ya boss. https://deploy-preview-271--crossplane.netlify.app/contribute/

![Screen Shot 2023-01-24 at 1 40 10 PM](https://user-images.githubusercontent.com/10537576/214380053-597c3fe0-c2bb-4807-b856-8711370b0ac5.png)